doc/kubebuilder-migration/project_migration_guide.md

```diff
@@ -0,0 +1,374 @@
+		return reconcile.Result{Requeue: true}, nil
+	} else if err != nil {
+		reqLogger.Error(err, "Failed to get Deployment")
+		return reconcile.Result{}, err
+	}
+
+	// Ensure the deployment size is the same as the spec
+	size := memcached.Spec.Size
+	if *found.Spec.Replicas != size {
+		found.Spec.Replicas = &size
+		err = r.Client.Update(context.TODO(), found)
+		if err != nil {
+			reqLogger.Error(err, "Failed to update Deployment", "Deployment.Namespace", found.Namespace, "Deployment.Name", found.Name)
+			return reconcile.Result{}, err
+		}
+		// Spec updated - return and requeue
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	// Update the Memcached status with the pod names
+	// List the pods for this memcached's deployment
+	podList := &corev1.PodList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(memcached.Namespace),
+		client.MatchingLabels(labelsForMemcached(memcached.Name)),
+	}
+	if err = r.Client.List(context.TODO(), podList, listOpts...); err != nil {
+		reqLogger.Error(err, "Failed to list pods", "Memcached.Namespace", memcached.Namespace, "Memcached.Name", memcached.Name)
+		return reconcile.Result{}, err
+	}
+	podNames := getPodNames(podList.Items)
+
+	// Update status.Nodes if needed
+	if !reflect.DeepEqual(podNames, memcached.Status.Nodes) {
+		memcached.Status.Nodes = podNames
+		err := r.Client.Status().Update(context.TODO(), memcached)
+		if err != nil {
+			reqLogger.Error(err, "Failed to update Memcached status")
+			return reconcile.Result{}, err
+		}
+	}
+	return ctrl.Result{}, nil
+}
+```
+
+## Generate CRDs
+
+Run [`make manifests`][generate_crd] to generate CRD manifests. They would be generated inside the `config/` folder.
+
+```sh
+$ make manifests
+go: creating new go.mod: module tmp
+go: finding sigs.k8s.io v0.2.5
+go: finding sigs.k8s.io/controller-tools/cmd v0.2.5
+go: finding sigs.k8s.io/controller-tools/cmd/controller-gen v0.2.5
+/Users/admin/go/bin/controller-gen "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+```
+
+## Operator Manifests
+
+### Operator deployment manifests
+
+The operator deployment manifest [`operator.yaml`][deployment_yaml] present inside `deploy/` folder in SDK project can be copied to `config/manager/manager.yaml`
+
+**Note**: 
+The kustomize file requires the operator deployment manifest to have the field `namespace` which is missing in the current operator.yaml of sdk project.
+
+### RBAC permissions
+
+The RBAC related yaml files present in `config/rbac/` of kubebuilder project are generated through the [rbac markers][rbac_markers] present in `controller/memcached_controller.go`.
+
+In our example of memcached operator, add the following rbac markers in `memcached_controller.go`:
+
+```Go
+// +kubebuilder:rbac:groups=apps,resources=deployments;pods;daemonsets;replicasets;statefulsets,verbs=get;update;patch;list;create;delete;watch
+```
```

Comment:
We should say where to put these RBAC markers (e.g. at the top of the file? As a comment on the `Reconcile` method?), and include some context around the markers in the example.